graphql/resolvers/exerciseSubmissionCrud.ts

```diff
@@ -0,0 +1,46 @@
+import { MutationAddExerciseSubmissionArgs } from '..'
+import { Context } from '../../@types/helpers'
+import prisma from '../../prisma'
+
+export const exerciseSubmissions = (
+  _parent: void,
+  _args: void,
+  context: Context
+) => {
+  const userId = context.req.user?.id
+  if (!userId) return []
+
+  return prisma.exerciseSubmission.findMany({
+    include: {
+      exercise: { include: { module: { include: { lesson: true } } } },
```

Comment:
I am currently not using the module or lesson data for the exercise, all I'm currently using is the exercise's ID, so maybe I can remove this to get a possible performance improvement. Or we can leave it like this if it doesn't make much of a difference and if this keeps things more simple.

Reply:
If it's not used, I think it should be removed. Then It'll be added once needed.

Reply:
Are there any plans to use module or lesson data in this resolver? If we know we need it soon for some other feature, I'd say we can keep the logic since we would need to add it soon anyway. If there's no set plans that require module or lesson data in this resolver, then i'd say remove it

Reply:
Here are all the possible use cases I can think of for exercise submissions:
1. Find all exercise submissions of the currently-logged-in user
2. Find all exercise submissions of another user
3. Find all exercise submissions of an exercise (by many different users)

The only use case I think we have plans for at the moment is #1, which this pull request covers. I can't think of a situation when we would need module or lesson data from the exercise submission.

The problem is that if I were to remove the module and lesson data, then it would create a situation where the GraphQL queries are not type-safe. To make it type-safe, I would have to change the GraphQL typedefs to match the type that I return from the resolver.

The current way we define ExerciseSubmission in the typedef file is:
```graphql
  type ExerciseSubmission {
    id: Int!
    user: User!
    exercise: Exercise!
    userAnswer: String!
  }
```
As you can see, we have to include an entire User and Exercise type from the resolver for this to be type-safe because someone might use those fields in the future and expect them to all be there and they would get a GraphQL validation error if they try to select a field that doesn't get returned.

To fix this, we would have to remove the unused data from the typedef so its type would look more like this:
```graphql
  type ExerciseSubmission {
    id: Int!
    userId: Int!
    exerciseId: Int!
    userAnswer: String!
  }
```
So I think I'll make that change since it seems like we don't need any of the other data from Exercise and User.

Reply:
Done